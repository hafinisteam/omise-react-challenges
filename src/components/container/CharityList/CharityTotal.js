import React from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import { UIButton } from "~/components/presentation/ui/button";
import COLOR from "~/utilities/layout/color";
import { donationDataSelector } from "~/state/ducks/donationData";

const Wrapper = styled.div`
  text-align: right;
`;

const TotalStyled = styled(UIButton)`
  border-color: ${COLOR.pink};
  color: ${COLOR.pink};
  cursor: default;
  &:hover {
    background: ${COLOR.white};
    color: ${COLOR.pink};
  }
`;

const CharityTotal = ({ total }) => {
  return (
    <Wrapper>
      <TotalStyled as="span">Total donation: {total} $</TotalStyled>
    </Wrapper>
  );
};

export default connect(state => ({
  total: donationDataSelector.getDonation(state)
}))(CharityTotal);
