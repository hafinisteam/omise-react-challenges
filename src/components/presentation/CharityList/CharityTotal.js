import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import Proptypes from 'prop-types';
import { UIButton } from '~/components/presentation/ui/button';
import COLOR from '~/utilities/layout/color';
import { donationDataSelector } from '~/state/ducks/donationData';
import { Modal, Empty, Tag } from 'antd';

const Wrapper = styled.div`
	text-align: right;
`;

const TotalStyled = styled(UIButton)`
	border-color: ${COLOR.pink};
	color: ${COLOR.pink};
	cursor: default;
	&:hover {
		background: ${COLOR.pink};
	}
`;

const PaymentItem = styled.div`
  border-bottom: 1px solid ${COLOR.gray};
  padding-bottom: 10px;
  margin-bottom: 9px;
  &:first-child {
    margin-top: 10px;
  }
  &:last-child {
    border-bottom: none;
    padding-bottom: 0;    
  }
	.name {
		font-size: 16px;
    color: ${COLOR.blue};
    margin-bottom: 0;
  }
  .info {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
`;

const paymentsInfo = (payments, charities) => {
	Modal.info({
		title: 'Donation payments list',
		content: payments.length ? (
			payments.map(p => {
				const charity = charities.find(c => p.charitiesId === c.id);
				return (
					<PaymentItem key={p.id}>
						<p className="name">{charity.name}</p>
            <div className='info'>
              <b>{p.amount}</b>
              <Tag color="geekblue">{p.currency}</Tag>
            </div>
					</PaymentItem>
				);
			})
		) : (
			<Empty />
		)
	});
};

const CharityTotal = ({ total, payments, charities }) => {
	return (
		<Wrapper>
			<TotalStyled onClick={() => paymentsInfo(payments, charities)}>
				Total donation: {total}
			</TotalStyled>
		</Wrapper>
	);
};

CharityTotal.propTypes = {
	total: Proptypes.number.isRequired,
	payments: Proptypes.array,
	charities: Proptypes.array
};

export default connect(state => ({
	total: donationDataSelector.getDonation(state),
	payments: donationDataSelector.getPayments(state),
	charities: donationDataSelector.getCharities(state)
}))(CharityTotal);
