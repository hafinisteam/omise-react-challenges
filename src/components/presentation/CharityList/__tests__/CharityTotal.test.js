import "jest-styled-components";
import renderer from "react-test-renderer";
import { CharityTotal } from "../CharityTotal";
import sinon from "sinon";

describe("<CharityTotal />", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<CharityTotal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("close button should call click events", () => {
    const onButonClick = sinon.spy();
    const wrapper = shallow(<CharityTotal payments={[]}  />);
    wrapper.find("TotalButton.open-modal-btn").simulate("click");
    expect(onButonClick).toHaveProperty("callCount", 0);
  });
});
