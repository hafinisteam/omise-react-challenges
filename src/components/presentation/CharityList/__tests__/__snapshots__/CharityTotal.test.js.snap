// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharityTotal /> should render correctly 1`] = `
.c0 {
  text-align: right;
}

.c1 {
  border: 1px solid #4674EE;
  border-radius: 3px;
  background: white;
  color: #4674EE;
  font-size: 16px;
  padding: 5px 10px;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  font-weight: 600;
  border-color: #ee468e;
  color: #ee468e;
  cursor: default;
}

.c1:hover {
  color: white;
  background: #4674EE;
}

.c1:hover {
  background: #ee468e;
}

<div
  className="c0"
>
  <button
    className="open-modal-btn c1"
    onClick={[Function]}
  >
    Total donation: 
    0
  </button>
</div>
`;
